packages/core/botpress/src/server/api.js

```diff
@@ -188,17 +188,18 @@ bp.createShortlink('chat', '/lite', {
     app.get(`/s/:name`, (req, res) => {
       const name = req.params.name.toLowerCase()
       const query = qs.stringify(req.query)
+      let link = links[name]
 
-      if (!links[name]) {
+      if (!link) {
         return res.status(404).send({ error: `Shortlink "${name}" not registered` })
       }
 
       if (query) {
-        const hasQuery = /\?/g.test(links[name])
-        links[name] = links[name].concat(`${hasQuery ? '&' : '?'}${query}`)
+        const hasQuery = /\?/g.test(link)
+        link = link.concat(`${hasQuery ? '&' : '?'}${query}`)
       }
 
-      res.redirect(links[name])
+      res.redirect(link)
```

Comment:
@nnn3d , trying to get what has changed here. Did you just replace `links[name]` via `link` everywhere while `link = links[name]`? :smile:

Reply:
@epaminond yes :) because `links` - [global constant](https://github.com/botpress/botpress/pull/820/files#diff-55ea99799cab87043e8c079cf8e94acaL153), and when to [`links[name]` will be added some string](https://github.com/botpress/botpress/pull/820/files#diff-55ea99799cab87043e8c079cf8e94acaL198) (added query params to end of string in this case) - it will be added to global variable again and again (see example in first message).  
`link` - local variable, and when works with `link`, 'links[name]' will not be rewritten.